clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/InternalLocalBootstrappingVeniceChangelogConsumer.java

```diff
@@ -372,14 +384,51 @@ protected <T> T processRecordBytes(
       PubSubTopicPartition partition,
       int readerSchemaId,
       long recordOffset) {
+    BootstrapState bootstrapState = bootstrapStateMap.get(partition.getPartitionNumber());
+    // In CATCHING_UP phase, there can be lots of updates to the same key.
+    // To improve bootstrapping performance, we'll put raw data in a temp cache (hashmap) and will
+    // process save to underlying storage on CATCHING_UP completion. Besides, we'll also skip RocksDB syncOffset.
+    if (bootstrapState.bootstrapState == PollState.CATCHING_UP) {
+      bootstrapState
```

Comment:
idk about this.  It seems like the cache could grow to a pretty large size.  We should probably put a size safety check in here to control flushing to disk.